pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -346,31 +350,70 @@ def build_model(
             )
 
             if self.time_varying_intercept:
-                time_index = pm.Data(
-                    "time_index",
+                intercept_time_index = pm.Data(
+                    "intercept_time_index",
                     self._time_index,
                     dims="date",
                 )
-                intercept_dist = get_distribution(
+
+                intercept_distribution = get_distribution(
                     name=self.model_config["intercept"]["dist"]
                 )
-                intercept = create_time_varying_intercept(
-                    time_index,
-                    self._time_index_mid,
-                    self._time_resolution,
-                    intercept_dist,
-                    self.model_config,
+                base_intercept = intercept_distribution(
+                    name="base_intercept", **self.model_config["intercept"]["kwargs"]
+                )
+
+                intercept_latent_process = create_time_varying_gp_multiplier(
+                    name="intercept",
+                    dims="date",
+                    time_index=intercept_time_index,
+                    time_index_mid=self._time_index_mid,
+                    time_resolution=self._time_resolution,
+                    model_config=self.model_config,
+                )
+                intercept = pm.Deterministic(
+                    name="intercept",
+                    var=base_intercept * intercept_latent_process,
+                    dims="date",
```

Comment:
I really like this idea, and I think you're right. However, I think we could proceed with the first implementation and take some time in the next release to give this module more shape.

Why? My idea is that at some point, in the same way that users can now define which variable is hierarchical or not, or centered hierarchical or not, they can define which variable will be time-dependent, and surely that would require changing and giving it greater flexibility to the tvp module.

Currently we only allow two variables to be time dependent, ideally the user could decide which one varies. The planned release `0.8.0` is pointing in that direction and I think it would be the ideal place to spend some time thinking about and adding this appropriately!

For now, perhaps this allows us to have user insights, and begin to show the difference between having or not having time components!